src/showingpreviously/cinemas/omniplex.py

```diff
@@ -0,0 +1,97 @@
+from datetime import datetime, timedelta
+from bs4 import BeautifulSoup
+import re
+
+import showingpreviously.requests as requests
+from showingpreviously.model import ChainArchiver, CinemaArchiverException, Chain, Cinema, Screen, Film, Showing
+from showingpreviously.consts import STANDARD_DAYS_AHEAD, UK_TIMEZONE, UNKNOWN_FILM_YEAR, UNKNOWN_SCREEN
+
+
+CINEMAS_URL = 'https://www.omniplex.ie/'
+SHOWINGS_URL = 'https://www.omniplex.ie/cinema/{cinema_id}'
+SCREEN_NAME_PATTERN = re.compile(r'var\s+screen\s*=\s*\'(?P<screen_name>.+?)\';')
+
+CHAIN = Chain('Omniplex Cinemas')
+
+
+def get_response(url: str) -> requests.Response:
+    r = requests.get(url)
+    if r.status_code != 200:
+        raise CinemaArchiverException(f'Got status code {r.status_code} when fetching URL {url}')
+    return r
+
+
+def get_cinemas_as_dict() -> dict[str, Cinema]:
+    r = get_response(CINEMAS_URL)
+    soup = BeautifulSoup(r.text, features='html.parser')
+    cinema_select = soup.find('select', {'id': 'homeSelectCinema'})
+    cinemas = {}
+    for cinema_option in cinema_select.find_all('option', {'id': True, 'data-thissite': True}):
+        cinema_name = cinema_option.text.strip()
+        cinema_id = cinema_option['id']
+        cinemas[cinema_id] = Cinema(cinema_name, UK_TIMEZONE)
+    return cinemas
+
+
+def get_attributes(at: str) -> dict[str, any]:
+    at = at.lower()
+    attributes = {'format': []}
+    if 'subtitled' in at:
+        attributes['subtitled'] = True
+    if 'sensory' in at:
+        attributes['format'].append('sensory')
+    if len(attributes['format']) == 0:
+        del attributes['format']
+    return attributes
+
+
+def get_showing_screen(booking_link: str) -> Screen:
+    r = get_response(booking_link)
+    if 'Booking Error' in r.text:
+        return UNKNOWN_SCREEN
+    screen_name = SCREEN_NAME_PATTERN.search(r.text).group('screen_name')
+    screen = Screen(screen_name)
+    return screen
+
+
+def get_showings_date(cinema_id: str, cinema: Cinema, dates: [str]) -> [Showing]:
+    url = SHOWINGS_URL.format(cinema_id=cinema_id)
+    r = get_response(url)
+    soup = BeautifulSoup(r.text, features='html.parser')
+    showings = []
+    for cinema_block in soup.find_all('div', {'class': 'rightHolder'}):
+        film_title = cinema_block.find('h3').text
+        film = Film(film_title, UNKNOWN_FILM_YEAR)
```

Comment:
The cinema does expose the name of the director, and the lead actors. If we were to go down the path of "add metadata to easily date films with unknown years", this would be a good candidate cinema to do it with,

Reply:
It's a good idea, but needs significant changes to the archiver before this can be added.